### ISSUE TITLE
Add Batch-Norm

### ISSUE THREAD

**Issue by shb84** [shb84/JENN#3]
The code works well as it is, but might converge faster if batch norm was implemented. Feel free to contribute.
